torch/utils/data/dataset.py

```diff
@@ -317,7 +334,20 @@ def class_function(cls, enable_df_api_tracing, source_dp, *args, **kwargs):
     def __getstate__(self):
         if IterDataPipe.getstate_hook is not None:
             return IterDataPipe.getstate_hook(self)
-        return self.__dict__
+        state_dict = {}
+        for k, v in self.__dict__.items():
+            if callable(v):
```

Comment:
I choose to only apply the `serialize_fn` on callable objects here. Any objection to this?